### PR TITLE
refactor(ndt_scan_matcher, ndt_omp): move ndt_omp into ndt_scan_matcher

### PR DESCRIPTION
**Pull request by SakodaShintaro**
## Description

Currently, we are using the [ndt_omp](https://github.com/tier4/ndt_omp) package as an external library. We would like to release this package through ROS2 bloom, but a package with the same name, `ndt_omp`, has already been released.

Since `ndt_omp` is currently under TIER IV organization, adding the prefix “autoware_” to `ndt_omp` is not ideal.

As `ndt_omp` is a small library, we are considering merging it into the `ndt_scan_matcher` package.

## Related links

- https://github.com/tier4/ndt_omp/pull/68

## How was this PR tested?
* `logging_simulator` works the same as before
* [TIER IV internal deterministic test](https://github.com/tier4/localization_evaluator_tools/actions/runs/11024014991)

## Notes for reviewers

None.

## Interface changes

`ndt_scan_matcher` no longer depends on `ndt_omp`.

## Effects on system behavior

None.

### CODE REVIEW

File: localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE [hunk 1]
```diff
@@ -0,0 +1,25 @@
+BSD 2-Clause License
```
Comment (esteve) [inline localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE:1]: The path for the header file needs to be under `autoware/ndt_scan_matcher` according to our file location guidelines, so it'll be `autoware/ndt_scan_matcher/ndt_omp` if you want to keep `ndt_omp` separate from the rest of the code.

**Review by esteve — CHANGES_REQUESTED** [review]

File: localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE (hunk 1 above)
Comment (SakodaShintaro) [reply localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/8912/commits/5bac6f4fa3b17e961766d4bfbf61df121d1b4687

Comment (SakodaShintaro) [reply localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE]: I'm sorry, the following include guard becomes too long to pass pre-commit.ci.
https://github.com/autowarefoundation/autoware.universe/blob/4bf27bd60a9c6641acae298974a335657d6a949e/localization/autoware_ndt_scan_matcher/include/autoware/ndt_scan_matcher/ndt_omp/multigrid_pclomp/multi_voxel_grid_covariance_omp_impl.hpp#L52-L53

I will fix again.

Comment (esteve) [reply localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE]: In fact, given that `ndt_omp` is not going to be exported outside the `ndt_scan_matcher` package, it'd be preferrable to move the `ndt_omp` headers to `src/`

Comment (SakodaShintaro) [reply localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE]: Currently, `ndt_omp` includes a static tool to execute NDT for a rosbag. 
e.g.) <https://github.com/tier4/ndt_omp/tree/tier4/main/apps>
It is useful for testing the NDT internal behavior.

After moving the code into `ndt_scan_matcher`, we plan to implement such a tool in `autoware_tools` or somewhere similar. Therefore, we would like to leave the header files as they are, and I have moved the implementation files to src.
https://github.com/autowarefoundation/autoware.universe/pull/8912/commits/6773e83bb070187ff7c46a94f85319d073fcf159

**Comment by anhnv3991** [comment]
@SakodaShintaro The code looks fine but ndt_omp is currently being used in map_assessment_tool, too. Is the tier4/ndt_omp still remain after moving ndt_omp to Autoware?

**Comment by SakodaShintaro** [comment]
@anhnv3991 
Yes, I will plan to archive [tier4/ndt_omp](https://github.com/tier4/ndt_omp).
https://docs.github.com/en/repositories/archiving-a-github-repository/archiving-repositories

**Review by YamatoAndo — APPROVED** [review]

File: localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE (hunk 1 above)
Comment (SakodaShintaro) [reply localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE]: @esteve 
I have contacted Dr. Koide and confirmed that this pull request is okay to be merged.
Ando-san has also approved this, so if you approve it, I will merge it.

Comment (SakodaShintaro) [reply localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE]: @esteve friendly ping.

Comment (SakodaShintaro) [reply localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE]: @esteve friendly ping.

Comment (SakodaShintaro) [reply localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE]: @mitsudome-r 
@esteve -san has marked "request changes" on this PR. While I've implemented the requested fixes, I haven't received any further response. Could you force a merge?

**Review by esteve — APPROVED** [review]
@SakodaShintaro LGTM. Thanks for addressing my feedback

File: localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE (hunk 1 above)
Comment (esteve) [reply localization/autoware_ndt_scan_matcher/include/autoware/ndt_omp/LICENSE]: @SakodaShintaro sorry, I missed your comment. I've approved this PR, thanks for addressing my feedback.

**Review by esteve — CHANGES_REQUESTED** [review]
@SakodaShintaro sorry, I missed something in my review. All headers must live under `include/autoware/ndt_scan_matcher`, so `ndt_omp` should be in `include/autoware/ndt_scan_matcher/ndt_omp`

**Comment by SakodaShintaro** [comment]
@esteve 
I think the current directory structure is what you described, but am I misunderstanding something?

```
shintarosakoda@dpc2304005:~/autoware$ tree ./src/universe/autoware.universe/localization/autoware_ndt_scan_matcher/include/
./src/universe/autoware.universe/localization/autoware_ndt_scan_matcher/include/
└── autoware
    └── ndt_scan_matcher
        ├── hyper_parameters.hpp
        ├── map_update_module.hpp
        ├── ndt_omp
        │   ├── LICENSE
        │   ├── README.md
        │   ├── estimate_covariance.hpp
        │   ├── multi_voxel_grid_covariance_omp.h
        │   ├── multigrid_ndt_omp.h
        │   └── ndt_struct.hpp
        ├── ndt_scan_matcher_core.hpp
        └── particle.hpp

3 directories, 10 files
```

What is the correct structure?

**Comment by esteve** [comment]
@SakodaShintaro ah, sorry, I think I misread the changes. That folder structure looks good to me, thanks.

**Review by esteve — APPROVED** [review]

**Pull request merged**